esm_analysis/carbon.py

```diff
@@ -344,45 +414,54 @@ def regression_against_index(ds, index, psig=None):
                 ds = ds_terms_anomaly.sel(time=slice(str(y1), str(y2)))
                 ind = index.sel(time=slice(str(y1), str(y2)))
                 terms = regression_against_index(ds, ind)
-                sens = pco2_sensitivity.sel(time=slice(str(y1), str(y2)))
+                sens = sensitivity.sel(time=slice(str(y1), str(y2)))
                 res.append(terms * nanmean(sens))
-        terms_in_pCO2_units = xr.concat(res, dim='time').mean('time')
+        terms_in_decompose_units = xr.concat(res, dim='time').mean('time')
 
     if plot:
-        terms_in_pCO2_units.to_array().plot(
+        terms_in_decompose_units.to_array().plot(
             col='variable', cmap='RdBu_r', robust=True, **plot_kwargs
         )
     else:
-        return terms_in_pCO2_units
+        return terms_in_decompose_units
 
 
 @check_xarray(0)
-def spco2_decomposition(ds_terms, detrend=True, order=1, deseasonalize=False):
-    """Decompose oceanic surface pco2 in a first order Taylor-expansion.
+def decomposition(
+    ds_terms, detrend=True, order=1, deseasonalize=False, decompose='spco2'
+):
+    """Decompose spco2 or co2_flx_ocean in a first order Taylor-expansion.
 
     Args:
-        ds_terms (xr.Dataset): containing cmorized variables:
-                               spco2 [ppm]: ocean pCO2 at surface
-                               talkos[mmol m-3]: Alkalinity at ocean surface
-                               dissicos[mmol m-3]: DIC at ocean surface
-                               tos [C] : temperature at ocean surface
-                               sos [psu] : salinity at ocean surface
+        ds_terms (xr.Dataset): containing cmorized variables as required by
+                               sensitivities
         detrend (bool): If True, detrend when generating anomalies. Default to
                         a linear (order 1) regression.
         order (int): If detrend is true, the order polynomial to remove from
                      your time series.
         deseasonalize (bool): If True, deseasonalize when generating anomalies.
+        decompose (str): variable to be decomposed. Choose from:
+                         ['co2_flx_ocean', 'spco2']
 
     Return:
-        terms_in_pCO2_units (xr.Dataset): terms of spco2 decomposition
+        terms_in_decompose_units (xr.Dataset): terms of spco2 decomposition
```

Comment:
ˋˋˋsuggestion
terms_in_units_of_decompose (xr.object): terms of decomposition
ˋˋˋ